src/python/pants/bin/local_pants_runner.py

```diff
@@ -220,11 +221,22 @@ def set_start_time(self, start_time):
     self._run_start_time = start_time
     self._reporting.initialize(self._run_tracker, self._options, start_time=self._run_start_time)
 
+    target_specs = [self.normalise_spec(spec) for spec in self._options.target_specs]
+    self._run_tracker.run_info.add_info("specs", target_specs, stringify=False)
+
     # Capture a repro of the 'before' state for this build, if needed.
     self._repro = Reproducer.global_instance().create_repro()
     if self._repro:
       self._repro.capture(self._run_tracker.run_info.get_as_dict())
 
+  @staticmethod
+  def normalise_spec(spec):
+    if spec.endswith("::"):
```

Comment:
I would love to have this list expanded as part of the metrics. Is it realistic to have this method return a list of targets?

Maybe as an additional field, because I agree that keeping this information is also important.

Reply:
My concern here is that at least internally at Twitter, we run a lot of commands like `./pants filter-minimize ::` where the `::` expands to a whole lot of targets, so this would start posting a _lot_ of information for a fair few pants runs...

If this were just a log, I'd do some kind of truncation (["foo", "bar", and seven more]), but if you're using this for things like SQL filters or joins, needing to know that some data may be omitted is scary... But accordingly, I also wouldn't want to conditionally flag whether we set this or not, for the same reasons...

Maybe we can work out something clever as a follow-up :)